### ISSUE TITLE
Seemingly making NaN the lowest number only affects Min not Max.

### ISSUE THREAD

**Issue by edgarfgp** [dotnet/runtime#73240]
### Description

Seemingly making NaN the lowest number only affects Min not Max.

While contributing to https://github.com/dotnet/fsharp/issues/13207.  and based on @tannergooding comment https://github.com/dotnet/fsharp/issues/13207#issuecomment-1194417377

### Reproduction Steps

```csharp
Console.WriteLine((new double[] { double.NaN, 2.0, 3.0 }).Max());
Console.WriteLine((new double[] { 3.0, double.NaN, 2.0 }).Max());
Console.WriteLine((new double[] { 3.0, double.NaN, 2.0 }).Min());
Console.WriteLine((new double[] { 3.0, double.NaN, 2.0 }).Min());

```

### Expected behavior

NaN
NaN
NaN
NaN

### Actual behavior

3
3
NaN
NaN

**Comment by tannergooding** [comment]
The root issue is that `Max`/`Min` don't follow any documentable IEEE 754 behavior around `NaNs`.

There are effectively three behaviors we could consider:
1. `max`/`min`
2. `maxNumber`/`minNumber`
3. `totalOrder`

`max`/`min` always propagate `NaN` back and thus `Max(NaN, number)` and `Min(NaN, number)` will both return `NaN`. Likewise, `-0` is less than `+0`.

`maxNumber`/`minNumber` always propagate back a `number` and thus `MaxNumber(NaN, number)` and `MinNumber(NaN, number)` will both return `number`. Likewise, `-0` is less than `+0`.

`totalOrder` gives a "total ordering" for NaN in relation to other floating-point values such that it has a "proper" sort position:
* Just like with `Max`, `MaxNumber`, `Min`, and `MinNumber`: `-0` is less than `+0`
* `-NaN` is less than any number
* `+NaN` is greater than any number
* `+sNaN` is less than `+qNaN`
* `-sNaN` is greater than `-qNaN`

`Enumerable.Max` and `Enumerable.Min` currently most closely resembles `totalOrder` but considers any `NaN` to be less than everything. This not only differs from the default `Max`/`Min` implementation which leads to confusion to end users on how it behaves, but also means that it doesn't match a behavior we could reasonably explain to users who expect IEEE 754 compliance.

Updating it to actually match `totalOrder` while a breaking change, is likely not to be super impactful because the `NaN` users are most likely to encounter is `float.NaN` and `double.NaN`, both of which are a `-qNaN` (this includes what constant folding will normalize a result to).

Updating it to actually match `Max`/`Min` and then exposing `MaxNumber`/`MinNumber` equivalents would be "best design" if doing this from day 1 since it would've then matched what `INumber<T>.Max/MaxNumber/Min/MinNumber` do. But this has the highest chance to break some user depending on the handling of NaNs today.

**Comment by eiriktsarpalis** [comment]
Seems like something we don't need to address in .NET 7?

**Comment by tannergooding** [comment]
Right, its a bit late in the cycle to consider a break like this and the current behavior, although confusing, has been around for many years.

We should consider it for .NET 8 and if nothing else improve the documentation around the type to call out the handling of NaN.

**Comment by edgarfgp** [comment]
@eiriktsarpalis Will this make it to NET 8? Thanks in advance

**Comment by eiriktsarpalis** [comment]
Hi, it's unlikely we'd be able to get around to fixing it during this release. Happy to review any community contributions that try to address it though.
